src/common/Parameter.cpp

```diff
@@ -3138,38 +3156,38 @@ void Parameter::get_display(char *txt, bool external, float ef) const
                 u = displayInfo.absoluteUnit;
             }
 
-            snprintf(txt, TXT_SIZE, "%.*f %s", (detailedMode ? 6 : displayInfo.decimals),
-                     displayInfo.scale * f, u.c_str());
+            txt = fmt::format("{:.{}f} {:s}", displayInfo.scale * f,
+                              (detailedMode ? 6 : displayInfo.decimals), u);
 
             if (f >= val_max.f &&
                 (displayInfo.customFeatures & ParamDisplayFeatures::kHasCustomMaxString))
             {
-                strxcpy(txt, displayInfo.maxLabel, TXT_SIZE);
+                txt = displayInfo.maxLabel;
             }
 
             if (f <= val_min.f &&
                 (displayInfo.customFeatures & ParamDisplayFeatures::kHasCustomMinString))
             {
-                strxcpy(txt, displayInfo.minLabel, TXT_SIZE);
+                txt = displayInfo.minLabel;
             }
 
             if (f == val_default.f &&
                 (displayInfo.customFeatures & ParamDisplayFeatures::kHasCustomDefaultString))
             {
-                strxcpy(txt, displayInfo.defLabel, TXT_SIZE);
+                txt = displayInfo.defLabel;
             }
 
-            return;
+            return txt;
         }
         case ATwoToTheBx:
         {
             if (can_temposync() && temposync)
             {
                 std::string res =
                     tempoSyncNotationValue(displayInfo.tempoSyncNotationMultiplier * f);
-                snprintf(txt, TXT_SIZE, "%s", res.c_str());
+                txt = fmt::format("{:s}", res);
```

Comment:
This can just be txt = res or better just return res

Reply:
Actually it doesn't need that local str::string at all, can just assign tempoSyncNotationValue to txt then return txt. Seems cleaner!